crypto/param_build.c

```diff
@@ -12,11 +12,47 @@
 #include <openssl/err.h>
 #include <openssl/cryptoerr.h>
 #include <openssl/params.h>
+#include <openssl/types.h>
 #include "internal/cryptlib.h"
 #include "openssl/param_build.h"
 
+/*
+ * The number of OSSL_PARAM elements a builder will allow.
+ */
+#define OSSL_PARAM_BLD_MAX          25
```

Comment:
This seems to be a little bit too small (when used for example for the RSA multi-prime support) and as it is now allocated on heap it should be fine to raise it. Unfortunately structs with flexible arrays at the end are C99 construct.

Reply:
25 should be plenty for the short term -- MP RSA is limited to 5 primes maximum.

I'd prefer to rework the param builder to remove the limit entirely -- that's a larger task and time is short for 3.0.

Reply:
> 25 should be plenty for the short term -- MP RSA is limited to 5 primes maximum.

The providers/implementations/keymgmt/rsa_kmgmt.c supports up to 10 primes actually so that would be over the 25 limit for all the primes, exponents and coefficients.

> I'd prefer to rework the param builder to remove the limit entirely -- that's a larger task and time is short for 3.0.

If the OSSL_PARAM_BLD_new() api required a maximum and there was no "dynamic reallocation" support the change could be pretty minimal I suppose.

But all this is just a note. I do not block the PR.

Reply:
I bit the bullet and converted the build to use safestack.  Limit removed :)